eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -365,18 +370,49 @@ function ProcessReplace
     return $replaceMatrix
 }
 
-function ProcessImport([MatrixParameter[]]$matrix, [String]$selection, [Array]$nonSparseParameters, [Hashtable]$displayNamesLookup)
-{
+function ProcessEnvironmentVariableReferences([array]$matrix, $displayNamesLookup) {
+    $updatedMatrix = @()
+
+    foreach ($element in $matrix) {
+        if (!$element || [bool]$element.PSobject.Properties.name -notmatch "_permutation") {
+            continue
+        }
+        $updated = [MatrixParameter[]]@()
+
+        foreach ($perm in $element._permutation) {
+            # Iterate nested permutations or run once for singular values (int, string, bool)
+            foreach ($flattened in $perm.Flatten()) {
+                if ($flattened.Value?.GetType() -eq "".GetType() -and $flattened.Value.StartsWith("env:")) {
+                    $envKey = $flattened.Value.Replace("env:", "")
+                    $value = [System.Environment]::GetEnvironmentVariable($envKey) ?? ""
```

Comment:
To me the code is a bit less clear if you don't know how get-item works, and there could be issues in the future if the outer conditional gets removed, allowing file lookups, etc.

Reply:
I was just trying to avoid us needing to parse variables when PS already does that. In this case it is pretty simple so I'm not too worried about splitting and just getting the env variable directly. I'll leave it to you :)